linkedlist/src/main/java/ua/hillel/LinkedList.java

```diff
@@ -1,69 +1,185 @@
 package ua.hillel;
 
-/**
- * Implement doubly linked list that is based on {@link Node<T>} class (you should implement it as well).
- * This is a simplified version of {@link java.util.LinkedList}.
- */
+import java.util.NoSuchElementException;
+
 public class LinkedList<T> implements List<T> {
+    class Node<T> {
+        T data;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(T data) {
+            this.data = data;
+            this.prev = null;
+            this.next = null;
+        }
+    }
 
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public LinkedList() {
+        head = null;
+        tail = null;
+        size = 0;
+    }
 
     @Override
     public void add(T element) {
-
+        Node<T> newNode = new Node<>(element);
+        if (head == null) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(int index, T element) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == size) {
+            add(element);
+            return;
+        }
+        Node<T> newNode = new Node<>(element);
+        if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else {
+            Node<T> node = getNodeAtIndex(index);
+            newNode.next = node;
+            newNode.prev = node.prev;
+            node.prev.next = newNode;
+            node.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> node = getNodeAtIndex(index);
```

Comment:
Does it return the correct element? My doubt is about the condition of the method `getNodeAtIndex`. You're looping inside that method, but the condition looks like this: `for (int i = 0; **i < index**; i++) `, and you need to find the node at position `index`. Inside this loop, it shouldn't be reached, because `i < index`. But you need a node at position `i <= index`, so this means you missed 1 iteration to the node you're looking for. In other words, you're operating over the `previous` element, not the one you need. Could you please check this?